src/pkg/clouds/aws/ecs/run.go

```diff
@@ -15,30 +16,37 @@ import (
 
 const taskCount = 1
 
-func (a *AwsEcs) Run(ctx context.Context, env map[string]string, cmd ...string) (TaskArn, error) {
-	// a.Refresh(ctx)
-
+func (a *AwsEcs) PopulateVPCandSubnetID(ctx context.Context, vpcID, subnetID string) error {
 	cfg, err := a.LoadConfig(ctx)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	if a.SubNetID == "" {
-		// Get a subnet ID
-		subnetsOutput, err := ec2.NewFromConfig(cfg).DescribeSubnets(ctx, &ec2.DescribeSubnetsInput{
-			Filters: []ec2types.Filter{
-				{
-					Name:   ptr.String("vpc-id"),
-					Values: []string{a.VpcID},
-					// 		Name:   ptr.String("map-public-ip-on-launch"),
-					// 		Values: []string{"true"},
-				},
-			},
-		})
+	if vpcID != "" && subnetID == "" {
```

Comment:
this code can be simplified by mutating the args (setting the one that's `""`) and having a single `return err` at the end

Reply:
Good point